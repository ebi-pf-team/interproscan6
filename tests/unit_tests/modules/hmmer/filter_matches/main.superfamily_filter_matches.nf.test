nextflow_process {

    name "Test Process SUPERFAMILY_FILTER_MATCHES"
    script "interproscan/modules/hmmer/filter/main.nf"
    process "SUPERFAMILY_FILTER_MATCHES"

    stage {
        symlink "interproscan/scripts/members/superfamily/parse_superfamily_out.py"
        symlink "tests/unit_tests/test_inputs/hmmer_filter_matches/superfamily/superfam_hmm.hmm"
        symlink "tests/unit_tests/test_inputs/hmmer_filter_matches/superfamily/superfamily_out_ass3_output"
    }

    test("Should run without failures") {

        when {
            params {}
            process {
                """
                input[0] = file("meta/tests/unit_tests/test_inputs/hmmer_filter_matches/superfamily/superfamily_out_ass3_output")
                input[1] = file("meta/tests/unit_tests/test_inputs/hmmer_filter_matches/superfamily/superfam_hmm.hmm")
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
