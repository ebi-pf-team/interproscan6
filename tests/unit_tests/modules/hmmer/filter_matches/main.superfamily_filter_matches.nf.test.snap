{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    "superfamily_parsed_superfamily_out_ass3_output.json:md5,6ffb4541862c5882ab2dd2f7a1c9fb0d"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-09-24T09:13:05.318168"
    }
}